proto/src/rr/rdata/caa.rs

```diff
@@ -733,15 +735,15 @@ impl KeyValue {
 ///      The length of the value field is specified implicitly as the
 ///      remaining length of the enclosing Resource Record data field.
 /// ```
-pub fn read(decoder: &mut BinDecoder, rdata_length: u16) -> ProtoResult<CAA> {
+pub fn read(decoder: &mut BinDecoder, rdata_length: Restrict<u16>) -> ProtoResult<CAA> {
     // the spec declares that other flags should be ignored for future compatability...
-    let issuer_critical: bool = decoder.read_u8()? & 0b1000_0000 != 0;
+    let issuer_critical: bool = decoder.read_u8()?.unverified() & 0b1000_0000 != 0;
 
-    let tag_len = decoder.read_u8()?;
-    let value_len = match rdata_length.checked_sub(u16::from(tag_len)).and_then(|l| l.checked_sub(2)) {
-        None => return Err("CAA tag character(s) out of bounds".into()),
-        Some(v) => v,
-    };
+    let tag_len = decoder.read_u8()?.unverified(); // verified usage below with checked sub
+    let value_len = rdata_length
+        .checked_sub(u16::from(tag_len))
+        .and_then(|l| l.checked_sub(2).ok_or(2))
+        .map_err(|_| ProtoError::from("CAA tag character(s) out of bounds"))?;
```

Comment:
@oherrala I think this is simpler than before with the new Restrict type. Thoughts?